server/channels/api4/file.go

```diff
@@ -513,31 +513,77 @@ func getFile(c *Context, w http.ResponseWriter, r *http.Request) {
 	defer c.LogAuditRec(auditRec)
 	model.AddEventParameterToAuditRec(auditRec, "force_download", forceDownload)
 
-	info, err := c.App.GetFileInfo(c.AppContext, c.Params.FileId)
-	if err != nil {
-		c.Err = err
-		setInaccessibleFileHeader(w, err)
+	fileInfos, storeErr := c.App.Srv().Store().FileInfo().GetByIds([]string{c.Params.FileId}, true, true)
+	if storeErr != nil {
+		c.Err = model.NewAppError("getFile", "api.file.get_file_info.app_error", nil, "", http.StatusInternalServerError)
+		setInaccessibleFileHeader(w, c.Err)
+		return
+	} else if len(fileInfos) == 0 {
+		c.Err = model.NewAppError("getFile", "api.file.get_file_info.app_error", nil, "", http.StatusNotFound)
+		setInaccessibleFileHeader(w, c.Err)
 		return
 	}
-	model.AddEventParameterAuditableToAuditRec(auditRec, "file", info)
 
-	channel, err := c.App.GetChannel(c.AppContext, info.ChannelId)
+	fileInfo := fileInfos[0]
+
+	channel, err := c.App.GetChannel(c.AppContext, fileInfo.ChannelId)
 	if err != nil {
 		c.Err = err
 		return
 	}
-	perm := c.App.SessionHasPermissionToReadChannel(c.AppContext, *c.AppContext.Session(), channel)
-	if info.CreatorId == model.BookmarkFileOwner {
-		if !perm {
-			c.SetPermissionError(model.PermissionReadChannelContent)
+
+	isContentReviewer := false
+	asContentReviewer, _ := strconv.ParseBool(r.URL.Query().Get(model.AsContentReviewerParam))
+	if asContentReviewer {
+		requireContentFlaggingEnabled(c)
```

Comment:
Would checking the feature is enabled conflict if there is a review in progress and then the feature is disabled, they won't be able to download the file?

Reply:
Done https://mattermost.atlassian.net/browse/MM-66670